packages/wallet/src/redux/protocols/withdrawing/readme.md

```diff
@@ -0,0 +1,34 @@
+# Withdrawing Protocol
+
+The purpose of this protocol is to withdraw funds from a directly funded channel.
+It covers:
+
+- Checking that a channel is closed.
+- Displaying to the user the funds allocated to them in the channel,
+- Getting approval from the user to proceed
+- Sending the withdraw + conclude to the blockchain
+- Getting acknowledgement from the user when done
+
+Out of scope (for the time being):
+
+- Retrying a transaction on failure.
+- Displaying an error message to the user on transaction failure.
+- Checking if a finalized outcome exists on chain(This will probably require a new saga).
```

Comment:
I was thinking about this because we need it for challenging too. It might make sense to have a section of the state where we just mirror the channel state from the adjudicator for channels we're interested in. That would mean storing the "funds held", "outcome" (if one exists) and "open challenge" and having a saga to keep them synced.

Reply:
That makes sense to me.

The other option of calling the contract when we need to check something seems slower and more painful to handle(since the calls will be async).

It sounds like we will need a saga that will:
 - take an adjudicator event
 - update channel state.
- possibly start a new process (IE: challenge detected would start the responding process)

Maybe we should create a card for this?

Reply:
https://github.com/magmo/apps/issues/384